org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/QueueViews.java

```diff
@@ -1,149 +1,149 @@
-	 */
-	public static String createSecondaryId(String uri, final String beanBundleName, final String beanClassName, final String queueName, final String topicName, final String submissionQueueName) {
-		
-		final StringBuilder buf = new StringBuilder();
-		if (uri!=null) {
-			try {
-				uri = URLEncoder.encode(uri, "UTF-8");
-			} catch (UnsupportedEncodingException e) {
-				e.printStackTrace(); // Not fatal
-			} 
-			append(buf, "uri",      uri);
-		}
-		append(buf, "beanBundleName",      beanBundleName);
-		append(buf, "beanClassName",       beanClassName);
-		append(buf, "queueName",           queueName);
-		append(buf, "topicName",           topicName);
-		append(buf, "submissionQueueName", submissionQueueName);
-		return buf.toString();
-	}
-	
-
-	protected static String createSecondaryId(String uri, String requestName, String responseName) {
-		final StringBuilder buf = new StringBuilder();
-		if (uri!=null) {
-			try {
-				uri = URLEncoder.encode(uri, "UTF-8");
-			} catch (UnsupportedEncodingException e) {
-				e.printStackTrace(); // Not fatal
-			} 
-			append(buf, "uri",      uri);
-		}
-		append(buf, "requestName",  requestName);
-		append(buf, "responseName", responseName);
-		return buf.toString();
-	}
-
-
-	protected static void append(StringBuilder buf, String name, String value) {
-		buf.append(name);
-		buf.append("=");
-		buf.append(value);
-		buf.append(";");
-	}
-
-}
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.scanning.api.event.queues;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.eclipse.scanning.api.event.EventConstants;
+
+/**
+ * A class to manage the e3 ids of views which will look at the queue.
+ * At the moment the connection options for which queue to open are
+ * configured into the secondary id for the view.
+ *
+ * A better design in future would be to have a service which manages
+ * the connection options and secondary id simply becomes a name for the
+ * options in the service. For instance 'ScanQueue' might give a StatusQueueView
+ * looking at the control machine messaging URL, looking for ScanBeans and using
+ * the default scan queue name and topic.
+ *
+ * @author Matthew Gerring
+ *
+ */
+public class QueueViews {
+
+	/**
+	 *
+	 * @return
+	 */
+	public static final String getQueueViewID() {
+		return "org.eclipse.scanning.event.ui.queueView"; // Might need to move this to a preference or other property.
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3400)